packages/features/bookings/lib/handleConfirmation.ts

```diff
@@ -66,6 +68,38 @@ export async function handleConfirmation(args: {
   const results = scheduleResult.results;
   const metadata: AdditionalInformation = {};
 
+  const eventType = booking.eventType;
+
+  const eventTypeWorkflows = eventType?.workflows?.map((workflowRel) => workflowRel.workflow) ?? [];
+
+  const teamId = await getTeamIdFromEventType({
+    eventType: {
+      team: { id: eventType?.teamId ?? null },
+      parentId: eventType?.parentId ?? null,
+    },
+  });
+
+  const triggerForUser = !teamId || (teamId && eventType?.parentId);
+
+  const userId = triggerForUser ? booking.userId : null;
+
+  const orgId = await getOrgIdFromMemberOrTeamId({ memberId: userId, teamId });
+
+  const isManagedEventType = !!eventType?.parentId;
+
+  const eventTypeMetadata = EventTypeMetaDataSchema.parse(eventType?.metadata || {});
+
+  const workflowsLockedForUser = isManagedEventType
+    ? !eventTypeMetadata?.managedEventConfig?.unlockedFields?.workflows
+    : false;
+
+  const workflows = await getAllWorkflows(
+    eventTypeWorkflows,
+    evt.organizer.id,
+    workflowsLockedForUser ? eventType?.parent?.teamId : teamId,
+    orgId
+  );
+
```

Comment:
This is some really important stuff. We should abstract it out and reuse here and in `handleCanceBooking` and `handleNewBooking` and maybe more places.

I think `getAllWorkflowsFromEventType` would be a good name. You give it an eventType and it does the rest.

Reply:
good call! done: https://github.com/calcom/cal.com/pull/15083/files#diff-5d5a2dd3d089d283853e724977b9ae82d83612101436022a8e6c6fe83d4d604fR72